### ISSUE TITLE
Helix Deployment script errors for "windows.11.amd64.android.open" Queue

### ISSUE THREAD

**Issue by parose1** [dotnet/dnceng#107]
- [x] This issue is blocking 
- [ ] This issue is causing unreasonable pain

This issue is blocking the deployment of Windows 11 machines into the "windows.11.amd64.android.open" Helix queue.

While trying to deploy new Windows 11 machines into Helix the Helix deployment script ran into a couple errors, and failed to complete. I have pasted screenshots of the errors below.

Test machine: DNCENGWIN-128
User: dotnet-bot
Access: can only be connected to via https://mlskvmui/ as these are on a isolated network that does not allow RDP

First error:
![image](https://github.com/dotnet/arcade/assets/48035740/9b00cc17-a4a9-410b-b955-2c9f7a7518ce)

Second error:
![image](https://github.com/dotnet/arcade/assets/48035740/15a199cc-ac37-44e6-869c-38f9e77aeac6)

I'm following the doc we have for the windows 10 connected android machines in our [OneNote here](https://microsoft.sharepoint.com/teams/DD_DDIT/DDITLabs/OneNote%20Library/ARM%20and%20Linux%20configuration/Windows.one#Mono%20deploy%20for%20HP%20SFF%20tethered%20to%20Android&section-id={30696A3C-CDDD-4BA5-B5F8-A4801FA7834C}&page-id={5008078D-5B0E-42A7-82BF-15F94FE2617C}&end)

Not sure if that needs to be changed for Windows 11

I left the CMD script window open on DNCENGWIN-128 in state for investigation.

### Release Note Category
- [ ] Feature changes/additions 
- [x] Bug fixes
- [ ] Internal Infrastructure Improvements
### Release Note Description

Adding allowable exit codes to the call to run vs_redist

**Comment by riarenas** [comment]
The first error looks like it was a network error when trying to contact the storage account. I tried running the script again and it didn't repro.

Currently looking into the VC redist errors.

**Comment by riarenas** [comment]
the VC redist installer was waiting for a reboot. I restarted the machine and then ran the script again and the only errors I saw were related to the runner user already existing and the windows defender exclusions already being there.

**Comment by riarenas** [comment]
@parose1 in order to diagnose whether this reboot is something that the OS was waiting for, or if we need to add a reboot to the artifact (can we add a reboot to the artifact in these on-prem scenarios?), would it be possible for you to run through your setup, but do a fresh reboot before running the setup script?

**Comment by parose1** [comment]
@riarenas sure I have some more fresh machines that I can test on. I'll try rebooting just before running the script and let you know how it goes. 

The windows 10 machines also error that the runner account already exists if you re-run the script, so we typically delete that account before a re-run. I will try that to if I have to re-run the script at all.

**Comment by parose1** [comment]
@riarenas I tried again on machine DNCENGWIN-129, rebooting right before running the Helix deployment script and verifying there was no pending reboots for updates. I still got the same VC redist error.

I then checked if the runner account was created but I did not see it in computer management so I rebooted and tried running the script again. This time I got errors for packages already existing in c:\helix-prep\. To try and get around this I deleted the following directories the script creates:
c:\cores
c:\h
c:\helix
c:\helix-prep
c:\osob

I rebooted again and tried running the script one more time but if failed for the error below:

![image](https://github.com/dotnet/dnceng/assets/48035740/d6161309-b870-4b96-8706-61508b84e1ea)

I have left DNCENGWIN-129 in that state if you want to take a look. I also setup DNCENGWIN-130 just before I would run the script if you need a fresh machine that hasn't had the script run yet to test on. Let me know if you need anything else to help investigate.

**Comment by riarenas** [comment]
Thanks @parose1 I think that's good info for me to continue! I'll check the machines out.

**Comment by riarenas** [comment]
I got a chance to look at the machines:

The debuggers error in -129 repros every time, but I wasn't able to repro it in -130 or when I looked at -128, so it makes me think this is something flaky with the machine or a side effect of one of the other failed artifacts.

For the clean machine, I followed these steps:

* I downloaded the VCRedist artifact and installed it manually: https://netcorenativeassets.blob.core.windows.net/resource-packages/external/windows/vc-redist/14.34/vc_redist.x64.exe and installed it the same way the artifact does it: with the /install /passive /norestart /log <path to where you want the logs> flags.
* After running this, I checked the installer again, and it was requesting a reboot to apply changes, I rebooted the machine.
* After this reboot, I ran the helix installation scripts, and the scripts finished running without errors and I was logged out of the machine and the runner account logged in and promptly kicked me from the kvm session.

I think this means that there is something going on with the windows-vc-redist artifact in this machine configuration. I'm unsure of the next steps here. It seems incorrect to download and install the vc redistributable before running the scripts because this is something we include in the queue configuration, but it also means we need a reboot in the middle of the execution of the scripts, even before we do the final reboot post artifact installation.
